moderncvstylecontemporary.sty

```diff
@@ -0,0 +1,55 @@
+%% start of file `moderncvstylecontemporary.sty'.
+%% Copyright 2006-2015 Xavier Danaux (xdanaux@gmail.com).
+%% Copyright 2023 Javier Lopez-Gomez (javier.lopez.gomez@proton.me).
+%
+% This work may be distributed and/or modified under the
+% conditions of the LaTeX Project Public License version 1.3c,
+% available at http://www.latex-project.org/lppl/.
+
+
+%-------------------------------------------------------------------------------
+%                identification
+%-------------------------------------------------------------------------------
+\NeedsTeXFormat{LaTeX2e}
+\ProvidesPackage{moderncvstylecontemporary}[2023/11/24 v2.0.0 modern curriculum vitae and letter style scheme: contemporary]
+
+% head section alignment options: "left" (default) or "right"
+\@initializecommand{\moderncvstyleheadoptions}{}
+\DeclareOption{left} {\edef\moderncvstyleheadoptions{\moderncvstyleheadoptions,left}}
+\DeclareOption{right}{\edef\moderncvstyleheadoptions{\moderncvstyleheadoptions,right}}
+\DeclareOption{qr}  {\edef\moderncvstyleheadoptions{\moderncvstyleheadoptions,qr}}
+\DeclareOption{noqr}{\edef\moderncvstyleheadoptions{\moderncvstyleheadoptions,noqr}}
+
+\DeclareOption*{}% avoid choking on unknown options
+\ExecuteOptions{left,qr}
+\ProcessOptions*\relax% \ProcessOptions* processes the options in the order provided (i.e., with the later ones possibly overriding the former ones), while \ProcessOptions processes them in the order of the package
+
+%-------------------------------------------------------------------------------
+%                fonts & icons
+%-------------------------------------------------------------------------------
+% Latin Modern fonts
+%\ifxetexorluatex
+%  \setmainfont{Latin Modern Roman}
+%  \setsansfont{Latin Modern Sans}
+%  \setmathfont{Latin Modern Math}
+%\else
+  \IfFileExists{lmodern.sty}%
+    {\RequirePackage{lmodern}}%
+    {}
+%\fi
+
+% symbols
+\moderncvicons{marvosym}
```

Comment:
Hi @LLdaniel,

> 2\. Taking the change here and risiking a bad color scheme because the symbols will be all grey then. I also will create a follow up issue to address the fact that it is difficult at the moment to change the icon color at different positions in the document. _The workaround would be to use grey symbols everywhere which does not exactly match the intended color scheme for "contemporary"._

Actually, I switched to `\moderncvicons{symbols}` and slightly changed `common/colors.tex` to define `default-socialicon-color` only if a previous definition is not available.
I also applied some some other minor changes to `moderncvstylecontemporary.sty` that were required after merging #151.

There seems to be a small issue regarding the coloring of the Signal icon, but seems totally unrelated to this PR.  Maybe doable in a different one (?).

Have a nice rest of the day!

Reply:
Hi @jalopezg-git, 
thanks a lot for your work!
> Actually, I switched to \moderncvicons{symbols} and slightly changed common/colors.tex to define default-socialicon-color only if a previous definition is not available. Now icons match the intended color scheme for 'contemporary'.

Great that the symbols are white now and match the overall style. One minor issue appears now that the symbols in the cover letter are also white. But I open a issue for that, because at the moment it is not possible to solve that for two locations at a time. Also for me the PR seems "big" enough^^

>I also applied some some other minor changes to moderncvstylecontemporary.sty that were required after merging https://github.com/moderncv/moderncv/pull/151.

Thanks in retrospective this was not the cleverest move from my side to integrate that PR first. At a first glance it seemed to me that we need that, but in fact it was more work for you... Sry.

>There seems to be a small issue regarding the coloring of the Signal icon, but seems totally unrelated to this PR. Maybe doable in a different one (?).

Yes this is more a matter of which kind of icon is picked up. Fontawesome 5 does not have a Signal icon and there is currently no Matrix icon as well. That is the reason why they are different. But this was **not** introduced with your PR, you are right. 

>Totally unrelated to the above, but 'contemporary' looks better with a much thinner header. Perhaps we could provide a template.tex that has most of the uncommon \social commented out + a suggestion to uncomment if needed.

Yeah, using the full list of icons is a bit overkill and hopefully nobody needs that in his/her CV. I know that it distorts the style this also happens for example with `\moderncvstyle[details]{classic}`.
For me it is ok to use show the full set of capable social icons with the knowledge that hopefully the full set is not used. But it would be also okay if we try to shorten the list and leave other icons only in the comments of `template.tex`.

Reply:
Hi @LLdaniel,

First of all, many thanks for quick the review :+1:!
 
> Great that the symbols are white now and match the overall style. One minor issue appears now that the symbols in the cover letter are also white. But I open a issue for that, because at the moment it is not possible to solve that for two locations at a time. Also for me the PR seems "big" enough^^

Oops, I did not notice when working on the patch!  Okay, let's address that in a follow-up PR!

> Thanks in retrospective this was not the cleverest move from my side to integrate that PR first. At a first glance it seemed to me that we need that, but in fact it was more work for you... Sry.

No worries!

> Yes this is more a matter of which kind of icon is picked up. Fontawesome 5 does not have a Signal icon and there is currently no Matrix icon as well. That is the reason why they are different. But this was **not** introduced with your PR, you are right.

Great!  I'll merge now and we can address this minor issues separately :slightly_smiling_face:.

> Yeah, using the full list of icons is a bit overkill and hopefully nobody needs that in his/her CV. I know that it distorts the style this also happens for example with `\moderncvstyle[details]{classic}`. For me it is ok to use show the full set of capable social icons with the knowledge that hopefully the full set is not used. But it would be also okay if we try to shorten the list and leave other icons only in the comments of `template.tex`.

Let's merge as-is and then we can change to a thinner header version, if need be.